FirebaseStorage/Sources/Internal/StorageFetcherService.swift

```diff
@@ -84,4 +80,22 @@ actor StorageFetcherService {
     }
     response(shouldRetry)
   }
+
+  /// Map of apps to a dictionary of buckets to GTMSessionFetcherService.
+  private actor FetcherServiceMap {
+    static let shared = FetcherServiceMap()
+
+    var fetcherServiceMap: [String: [String: GTMSessionFetcherService]] = [:]
+
+    func get(appName: String, bucket: String) -> GTMSessionFetcherService? {
+      return fetcherServiceMap[appName]?[bucket]
+    }
+
+    func set(appName: String, bucket: String, fetcher: GTMSessionFetcherService) {
+      if fetcherServiceMap[appName] == nil {
+        fetcherServiceMap[appName] = [:]
+      }
+      fetcherServiceMap[appName]?[bucket] = fetcher
```

Comment:
optional: I believe it works to do something like this:
```suggestion
      fetcherServiceMap[appName, default: [:]][bucket] = fetcher
```

Reply:
Nice! Done.